# Financial Analysis of Microsoft, Tesla, and Apple (10-K Data)

### Objective
The objective of this analysis is to extract key financial data from the 10-K filings of Microsoft, Tesla, and Apple for the last three fiscal years and analyze trends using Python. 

The analysis focuses on:
- Total Revenue
- Net Income
- Total Assets
- Total Liabilities
- Operating Cash Flow

The insights derived will demonstrate how structured financial data can support the development of an AI-powered financial chatbot.

### Data Extraction

Financial data was manually extracted from the SEC EDGAR database from the 10-K filings of:

- Microsoft
- Tesla
- Apple

For each company, data for the last three fiscal years (2023–2025) was collected from:
- Consolidated Income Statements
- Consolidated Balance Sheets
- Consolidated Statements of Cash Flows

The extracted data was compiled into an Excel file and converted to CSV format for analysis.

### IMPORT LIBRARIES

In [1]:
import pandas as pd

### LOAD DATA

In [2]:
df = pd.read_csv("C:/Users/LENOVO/OneDrive/Desktop/FORAGE-BOC-TASK/Data.csv")
df['Year'] = pd.to_numeric(df['Year'], errors='coerce')
df = df.dropna(subset=['Year'])
df['Year'] = df['Year'].astype(int)

#df
df.tail(10)


,Company,Year,Revenue (USD Millions),Net Income (USD Millions),Assets (USD Millions),Liabilities (USD Millions),Operating Cash Flow (USD Millions)
0,Microsoft,2025,281724.0,101832.0,619003.0,275524.0,136162.0
1,Microsoft,2024,245122.0,88136.0,512163.0,243686.0,118548.0
2,Microsoft,2023,211915.0,72361.0,411976.0,205753.0,87582.0
3,Tesla,2025,94827.0,3855.0,137806.0,54941.0,14747.0
4,Tesla,2024,97690.0,7153.0,122070.0,48390.0,14923.0
5,Tesla,2023,96773.0,14974.0,106618.0,43009.0,13256.0
6,Apple,2025,416161.0,112010.0,359241.0,285508.0,111482.0
7,Apple,2024,391035.0,93736.0,364980.0,308030.0,118254.0
8,Apple,2023,383285.0,96995.0,352583.0,290437.0,110543.0


In [3]:
df = df.sort_values(by=['Company', 'Year'])
df

,Company,Year,Revenue (USD Millions),Net Income (USD Millions),Assets (USD Millions),Liabilities (USD Millions),Operating Cash Flow (USD Millions)
8,Apple,2023,383285.0,96995.0,352583.0,290437.0,110543.0
7,Apple,2024,391035.0,93736.0,364980.0,308030.0,118254.0
6,Apple,2025,416161.0,112010.0,359241.0,285508.0,111482.0
2,Microsoft,2023,211915.0,72361.0,411976.0,205753.0,87582.0
1,Microsoft,2024,245122.0,88136.0,512163.0,243686.0,118548.0
0,Microsoft,2025,281724.0,101832.0,619003.0,275524.0,136162.0
5,Tesla,2023,96773.0,14974.0,106618.0,43009.0,13256.0
4,Tesla,2024,97690.0,7153.0,122070.0,48390.0,14923.0
3,Tesla,2025,94827.0,3855.0,137806.0,54941.0,14747.0


In [6]:
df['Revenue Growth (%)'] = df.groupby('Company')['Revenue (USD Millions)'].pct_change() * 100
df['Net Income Growth (%)'] = df.groupby('Company')['Net Income (USD Millions)'].pct_change() * 100
df['Asset Growth (%)'] = df.groupby('Company')['Assets (USD Millions)'].pct_change() * 100
df['Operating Cash Flow Growth (%)'] = df.groupby('Company')['Operating Cash Flow (USD Millions)'].pct_change() * 100
df.fillna(0, inplace=True)
df

,Company,Year,Revenue (USD Millions),Net Income (USD Millions),Assets (USD Millions),Liabilities (USD Millions),Operating Cash Flow (USD Millions),Revenue Growth (%),Net Income Growth (%),Asset Growth (%),Operating Cash Flow Growth (%)
8,Apple,2023,383285.0,96995.0,352583.0,290437.0,110543.0,0.000000,0.000000,0.000000,0.000000
7,Apple,2024,391035.0,93736.0,364980.0,308030.0,118254.0,2.021994,-3.359967,3.516052,6.975566
6,Apple,2025,416161.0,112010.0,359241.0,285508.0,111482.0,6.425512,19.495178,-1.572415,-5.726656
2,Microsoft,2023,211915.0,72361.0,411976.0,205753.0,87582.0,0.000000,0.000000,0.000000,0.000000
1,Microsoft,2024,245122.0,88136.0,512163.0,243686.0,118548.0,15.669962,21.800417,24.318650,35.356580
0,Microsoft,2025,281724.0,101832.0,619003.0,275524.0,136162.0,14.932156,15.539621,20.860546,14.858117
5,Tesla,2023,96773.0,14974.0,106618.0,43009.0,13256.0,0.000000,0.000000,0.000000,0.000000
4,Tesla,2024,97690.0,7153.0,122070.0,48390.0,14923.0,0.947578,-52.230533,14.492862,12.575438
3,Tesla,2025,94827.0,3855.0,137806.0,54941.0,14747.0,-2.930699,-46.106529,12.890964,-1.179388


### Financial Trend Analysis (2023–2025)
##### Microsoft

Revenue grew consistently in 2024 (15.67%) and 2025 (14.93%).

Net income increased significantly in 2024 (21.80%) and continued growing in 2025 (15.54%).

Assets and operating cash flow also showed strong double-digit growth.

This indicates strong operational efficiency and expanding business performance.

##### Tesla

Revenue growth remained relatively flat, with slight decline in 2025 (-2.93%).

Net income declined sharply in both 2024 (-52.23%) and 2025 (-46.11%).

However, assets continued growing, suggesting continued investment and expansion.

This may indicate margin pressure despite asset growth.

##### Apple

Revenue showed steady growth, especially in 2025 (6.43%).

Net income declined in 2024 but rebounded strongly in 2025 (19.50%).

Asset growth was moderate.

Operating cash flow remained stable overall.

### Methodology

Financial data was manually extracted from each company’s 10-K filings (Microsoft, Tesla, Apple).

Data was compiled into Excel and converted to CSV format.

Data was cleaned and sorted by Company and Year (ascending).

Year-over-year growth was calculated using pandas groupby() and pct_change() functions.

Growth rates were analyzed to identify financial trends across companies.

In [5]:
df.groupby('Company')[[
    'Revenue Growth (%)',
    'Net Income Growth (%)'
]].mean()

,Revenue Growth (%),Net Income Growth (%)
Company,,
Apple,4.223753,8.067605
Microsoft,15.301059,18.670019
Tesla,-0.991560,-49.168531


### Conclusion

Microsoft demonstrates the strongest consistent financial growth among the three companies, while Tesla shows volatility in profitability. Apple remains financially stable with moderate growth.